integration/kubernetes/helm-chart/alluxio/values.yaml

```diff
@@ -194,25 +194,35 @@ jobMaster:
 # UFS journal with HDFS example
 # journal:
 #   type: "UFS"
+#   ufsType: "non-local"
 #   folder: "hdfs://{$hostname}:{$hostport}/journal"
 # EMBEDDED journal to /journal example
 # journal:
 #   type: "EMBEDDED"
 #   folder: "/journal"
 journal:
-  type: "UFS" # "UFS" or "EMBEDDED"
-  ufsType: "local" # Ignored if type is "EMBEDDED". "local" or "HDFS"
-  folder: "/journal" # Master journal folder
+  # [ Required arguments ]
+  type: "UFS" # One of "UFS" or "EMBEDDED"
+  folder: "/journal" # Master journal directory or equivalent storage path
+  # [ Conditionally required arguments ]
+  #
+  # [[ UFS-backed journal options (journal.type="UFS") ]]
+  ufsType: "local" # One of "local" or "non-local"
+  #
+  # [[ K8s volume options ]]
+  # - required when using an EMBEDDED journal (journal.type="EMBEDDED")
+  # - required when using a local UFS journal (journal.type="UFS" and journal.ufsType="local")
+  #
   # volumeType controls the type of journal volume.
-  # It can be "persistentVolumeClaim" or "emptyDir"
-  volumeType: persistentVolumeClaim
+  volumeType: persistentVolumeClaim # One of "persistentVolumeClaim" or "emptyDir"
   size: 1Gi
   # Attributes to use when the journal is persistentVolumeClaim
   storageClass: "standard"
   accessModes:
     - ReadWriteOnce
   # Attributes to use when the journal is emptyDir
   medium: ""
+  # [ Optional arguments ]
```

Comment:
I like how you use `# [ Required arguments ]` as a block comment to separate the required/optional properties in groups. Let's keep using these.

Another option as someone suggested [here](https://stackoverflow.com/a/50037498/4933827) is to use a real text block instead of comments. Not sure if that can be an option here.

Reply:
I was actually thinking about refactoring the values here into sub-sections for each and using null-checks on fields for flow-control rather than having multiple flag values. So for desciptions we could have something like:
```
journal:
  ufs:
    description: "..."
    local:
      description: "..."
    hdfs:
      description: "..."
  embedded:
    description: "..."
  volumes:
    desciption: "..."
    persistentVolumeClaim:
    emptyDir:
```

I figured that type of refactor would be out-of-scope for this PR though (especially since it would change the user-facing values drastically). So I'll opt for just using YAML comments for now.